### PR TITLE
Fix Interpretability Methods `target_class_idx`

### PR DESCRIPTION
**Pull request by Logiquo**
1. Unify all interpretability to support `target_class_idx` and its semantics
2. `target_class_idx` becomes a no-op for binary classification, and select target class we want to interpret for multi-class and multi-label.
3. The multi-label will be selected based on `argmax` if `target_class_idx` is not provided. 
4. All samples (including negatives) will be evaluated in the interpretability metric, can support `sample_filter` which can specify POSITIVE, NEGATIVE and IGNORE based on the predicated probablity.

### CODE REVIEW

File: pyhealth/metrics/interpretability/evaluator.py [hunk 1]
```diff
@@ -30,17 +32,45 @@ class Evaluator:
...
+        ...     threshold_sample_filter,
+        ... )
         >>>
-        >>> # Initialize evaluator
+        >>> # Initialize evaluator with default filter
         >>> evaluator = Evaluator(model)
         >>>
+        >>> # Initialize with custom filter that ignores low-confidence
+        >>> def confident_filter(class_probs, classifier_type):
+        ...     batch_size = class_probs.shape[0]
```
Comment (jhnwu3) [inline pyhealth/metrics/interpretability/evaluator.py:66]: Just curious, how does this work? That's pretty cool though. Can we add examples here in the examples/interpretability/ directory that show how a user might do this for our binary cases here iirc?

File: pyhealth/interpret/methods/base_interpreter.py [hunk 2]
```diff
@@ -207,6 +211,42 @@ def attribute(
...
+        logit to explain.  All prediction modes share this single code
+        path:
+
+        * **Binary** (single logit): ``target_class_idx`` is a no-op
+          because there is only one output.  Always returns zeros
+          (index 0).
+        * **Multi-class / multi-label**: uses ``target_class_idx`` if
+          given, otherwise the argmax of logits.
+
+        Args:
```
Comment (jhnwu3) [inline pyhealth/interpret/methods/base_interpreter.py:231]: I like this approach.

**Comment by Logiquo** [comment]
image

image

We can have custom function to determine what sample is considered as positive/negative/ignore.

File: pyhealth/metrics/interpretability/evaluator.py (hunk 1 above)
Comment (Logiquo) [reply pyhealth/metrics/interpretability/evaluator.py:66]: Added
